form/src/main/java/org/dhis2/form/ui/FormViewModel.kt

```diff
@@ -766,6 +767,7 @@ class FormViewModel(
             try {
                 async(dispatcher.io()) {
                     repository.completeEvent()
+                    preferenceProvider.setValue(Preference.PREF_COMPLETED_EVENT, repository.getRecordUid())
```

Comment:
Why do we need to expose a value from valueStore(_data_) to ViewModel(_ui_) to store it in preferenceProvdier(_data_) ?. I was thinking for example... if it were a Use Case (ie: CompleteEvent), it can complete in SDK and then update preferences. As we don't want to do any refactor y the code freeze, I would set this reponsibility to the Repository.
What do you think to inject preferenceProvider in FormRepositoryImpl and in fun complete() do the valueStore complete plus the updating preferences?

Reply:
You are right, I wanted to avoid injecting the PreferenceProvider in another place as we already had access to it from the viewmodel, although we didn't have access to the record uid. I will also remove the PreferenceProvider calls from the viewmodel to the repositoryso that it's all in the same place.